File: Jour4/jour4.pas
program Permutation;
var
nbr1    : real;
nbr2    : real;
temp   : real;

begin
	writeln(' =================== Ce pregramme permute les valeurs de deux variables.  ===================');
	write('Donner la premiäre valeur : ');
	read(nbr1);
	//Attention aucun control sur la valeur saisie.

	write('Donner la deuxiäme valeur : ');
	read(nbr2);
	//Attention aucun control sur la valeur saisie

//--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
	writeln('********** PERMUTATION AVEC VARIABLE TEMPORAIRE ********** ');
	writeln(' Avant permutation : Nombre1 = ',nbr1:0:2,'  Nombre2 =  ',nbr2:0:2);
	temp := nbr1;
	nbr1 := nbr2;
	nbr2 := temp;
	writeln(' Apräs permutation : Nombre1 = ',nbr1:0:2,'  Nombre2 =  ',nbr2:0:2);

//--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
	writeln(' ********** PERMUTATION SANS VARIABLE TEMPORAIRE ********** ');
	writeln(' Avant permutation : Nombre1 = ',nbr1:0:2,'  Nombre2 =  ',nbr2:0:2);
	nbr1 := nbr1 + nbr2;
	nbr2 := nbr1 - nbr2;
	nbr1 := nbr1 - nbr2;
	writeln(' Avant permutation : Nombre1 = ',nbr1:0:2,'  Nombre2 =  ',nbr2:0:2);
//--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
end.

File: Jour4/jour4.c
#include<stdio.h>

int main()
{
	float nbr1,nbr2,temp;
	printf(" =================== Ce pregramme permute les valeurs de deux variables.  ===================\n");

	printf("Donner la premiere valeur : ");
	scanf("%f",&nbr1);
	//Attention aucun control sur la valeur saisie.

	printf("Donner la deuxieme valeur : ");
	scanf("%f",&nbr2);
	//Attention aucun control sur la valeur saisie

//-------------------------------------------------------------------------------------------------------
	printf(" ********** PERMUTATION AVEC VARIABLE TEMPORAIRE ********** \n");
	printf("\n Avant permutation : Nombre1 = %.2f  | Nombre2 = %.2f \n",nbr1,nbr2);
	temp = nbr1;
	nbr1 = nbr2;
	nbr2 = temp;
	printf("\n Apres permutation : Nombre1 = %.2f  | Nombre2 = %.2f \n",nbr1,nbr2);

//-------------------------------------------------------------------------------------------------------
	printf(" ********** PERMUTATION AVEC VARIABLE TEMPORAIRE ********** \n");
	printf("\n Avant permutation : Nombre1 = %.2f  | Nombre2 = %.2f \n",nbr1,nbr2);
	nbr1 += nbr2;
	nbr2 = nbr1 - nbr2;
	nbr1 = nbr1 - nbr2;
	printf("\n Apres permutation : Nombre1 = %.2f  | Nombre2 = %.2f \n",nbr1,nbr2);
//-------------------------------------------------------------------------------------------------------
	
	return 0;
}